src/services/env.ts

```diff
@@ -0,0 +1,73 @@
+import { PATH_CONFIG_DEFAULT } from '@/pathConfigDefault'
+import { PathConfig } from '@/types/index'
+
+export type EnvVars = {
+  KUMA_NAME: string
+  KUMA_VERSION: string
+  KUMA_BASE_PATH: string
+  KUMA_API_URL: string
+  KUMA_DOCS_URL: string
+  KUMA_UTM_QUERY_PARAMS: string
+  KUMA_FEEDBACK_URL: string
+  KUMA_CHAT_URL: string
+}
+export default class Env {
+  env: EnvVars
+  constructor(env: EnvVars) {
+    this.env = env
+  }
+
+  var(key: keyof EnvVars) {
+    return this.env[key]
+  }
+}
+
+// TODO(jc): for ease just do all this here for until we have a proper container
+const config = readPathConfigFromDom()
+const version = semver(config.version)
+
+export const env = new Env({
+  KUMA_NAME: `${import.meta.env.VITE_NAMESPACE}`,
+  KUMA_VERSION: version.pre,
+  KUMA_API_URL: config.apiUrl,
+  KUMA_BASE_PATH: config.baseGuiPath,
+  // TODO(jc): not totally sure we need to use a regex here, maybe just split and join if not
+  KUMA_DOCS_URL: `${import.meta.env.VITE_DOCS_BASE_URL}/${version.patch === '0.0.0' ? 'dev' : version.patch.replace(/\.\d+$/, '.x')}`,
+  // remove the ? for now incase we need to append to an already query
+  // param'ed URL
+  KUMA_UTM_QUERY_PARAMS: import.meta.env.VITE_UTM.substring(1),
+  KUMA_FEEDBACK_URL: import.meta.env.VITE_FEEDBACK_URL,
+  KUMA_CHAT_URL: import.meta.env.VITE_CHAT_URL,
```

Comment:
**Suggestion**: I think it would be better if none of these values were prefixed with `KUMA_` so that we minimize the chance for merge conflicts between this project and possible forks of it that won’t be called Kuma.

(Though I would use `PRODUCT_NAME` and `GUI_BASE_PATH`)

Reply:
I would prefer to prefix things really, especially is some of these variables are coming from your environment. Right now I'd prefer to get this merged as there's still work to be done on top of this. But another option would be `KM_` (i.e. KuMa). Again it's just naming really so happy either way as long as its prefixed. If you feel strongly about this now happy to change up to use `KM_` here, otherwise we can revisit later.

Reply:
It’s just that every time, you’d change a line with an `env` reference or an adjacent line and a fork has this value renamed, you’ll get a merge conflict. This can be completely avoided by not prefixing them with something product-specific.

Reply:
> Right now I'd prefer to get this merged as there's still work to be done on top of this.

In the interest of this, I would prefer to consider this PR with the scope of abstracting away static data access but without introducing the service container pattern. We clearly differ greatly in our opinions on that pattern and so there is need for further exchange.

Reply:
The product is always going to be a using this repository i.e. Kuma, even as a base. Whether the product is called Kuma or something else, naming of properties in code doesn't matter really, main thing is avoiding picking up environment variables that aren't for this project. Most projects that use environment variables use prefixes.

Reply:
> In the interest of this, I would prefer to consider this PR with the scope of abstracting away static data access but without introducing the service container pattern. We clearly differ greatly in our opinions on that pattern and so there is need for further exchange.

Ok but can we consider the naming subject resolved? Sounds like that from our discussions offline?

Reply:
Yes, sure.